### PR TITLE
fix(docs): Update the create a source plugin tutorial

### PR DESCRIPTION
**Pull request by KyleAMathews**
Did an initial pass on improving the tutorial. Some notes on additional changes I think should be made:

- we really need to make the mental model clear for what a source plugin is. Having talked to a number of people in process of building a source plugin, they're often missing understanding of even very basic things. So similar to the main tutorial we should have people go step-by-step through the whole sweep of the data layer e.g. the different actions, showing them the LMDB file on disk, etc.
- the example needs to a lot simpler — Apollo, etc. gets in the way of understanding things. Plus we should steer people towards Got
- the tutorial mixes basic & advanced features — we should separate the basic tutorial from advanced features like relationships, images, content sync
- make really clear how node caching works i.e. that nodes stay in the db until they're deleted (or not touched). This comes up over and over.
- relatedly: make really clear how to do delta syncs. This also is rarely understood and comes up over and over

### CODE REVIEW

**Review by DSchau — DISMISSED** [review]
Seems very sensible!

File: docs/docs/how-to/plugins-and-themes/creating-a-source-plugin.md [hunk 1]
```diff
@@ -5,25 +5,26 @@ tableOfContentsDepth: 2
...
 
-- Ensures local data is synced with its source and is 100% accurate.
-- Creates [nodes](/docs/reference/graphql-data-layer/node-interface/) with accurate media types, human-readable types, and accurate
-  [contentDigests](/docs/reference/graphql-data-layer/node-interface/#contentdigest).
-- Links nodes & creates relationships between them.
-- Lets Gatsby know when nodes are finished sourcing so it can move on to processing them.
+- Ensures the data in the Gatsby DB is synced with the latest updates from its source
+- Creates [nodes](/docs/reference/graphql-data-layer/node-interface/) (Gatsby's name for an object) with accurate media types, human-readable types, and accurate
+  [contentDigests](/docs/reference/graphql-data-layer/node-interface/#contentdigest)
+- Create relationships between nodes
```
Comment (DSchau) [inline docs/docs/how-to/plugins-and-themes/creating-a-source-plugin.md:21]:
```suggestion
- Creates relationships between nodes
```

File: docs/docs/how-to/plugins-and-themes/creating-a-source-plugin.md [hunk 2]
```diff
@@ -5,25 +5,26 @@ tableOfContentsDepth: 2
...
 
 A source plugin is a regular npm package. It has a `package.json` file, with optional dependencies, as well as a [`gatsby-node.js`](/docs/reference/config-files/gatsby-node/) file where you implement Gatsby's Node APIs. Read more about [files Gatsby looks for in a plugin](/docs/files-gatsby-looks-for-in-a-plugin/) or [creating a generic plugin](/docs/how-to/plugins-and-themes/creating-a-generic-plugin).
 
 ## Why create a source plugin?
 
-Source plugins convert data from any source into a format that Gatsby can process. Your Gatsby site can use several source plugins to combine data in interesting ways.
+Source plugins makes data from any source available to your Gatsby sites. Your Gatsby site can use several source plugins to combine data in interesting ways.
```
Comment (DSchau) [inline docs/docs/how-to/plugins-and-themes/creating-a-source-plugin.md:27]:
```suggestion
Source plugins make data from any source available to your Gatsby sites. Your Gatsby site can use several source plugins, like commerce data from Shopify, or content from one or more content management systems (like Contentful, WordPress, etc.), all in a unified graph.
```

Comment (DSchau) [reply docs/docs/how-to/plugins-and-themes/creating-a-source-plugin.md:27]: You can feel free to tweak the later part. I found "in interesting ways" kind of vague?

**Review by LekoArts — APPROVED** [review]

**Pull request merged**